iModelCore/ECDb/Tests/NonPublished/CommonTableExpTests.cpp

```diff
@@ -206,6 +206,144 @@ TEST_F(CommonTableExpTestFixture, MetaQuery) {
+                        FROM meta.ECClassDef c
+                            JOIN meta.ClassHasBaseClasses cbc ON cbc.SourceECInstanceId = c.ECInstanceId
+                            JOIN base_classes  ON aId = cbc.TargetECInstanceId
+                    ORDER BY 1
+                )
+            SELECT aId, aParentId, aDepth, aPath  from base_classes
+        ))";
+
+        ECSqlStatement stmt;
+        ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, query));
+        ASSERT_STREQ("aPath", stmt.GetColumnInfo(0).GetProperty()->GetName().c_str());
+        /*
+        aId aParentId aDepth aPath
+        --- --------- ------ --------
+        73    (null) 0      A
+        74        73 1      A/AA
+        75        74 2      A/AA/AAA
+        76        74 2      A/AA/AAB
+        77        73 1      A/AB
+        78        77 2      A/AB/ABA
+        79        77 2      A/AB/ABB
+        */
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AA", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AA/AAA", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AA/AAB", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AB", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AB/ABA", stmt.GetValueText(0)) << "aPath";
+
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("A/AB/ABB", stmt.GetValueText(0)) << "aPath";
+
+    }
+
+    // get all property in for a given class
+    if (true) {
+        auto query = R"(SELECT * FROM(
+            WITH RECURSIVE
+                base_classes (aId, aParentId, aPath, aDepth) AS (
+                    SELECT c.ECInstanceId, null, c.Name, 0  FROM meta.ECClassDef c WHERE c.Name=?
+                    UNION ALL
+                    SELECT c.ECInstanceId, cbc.TargetECInstanceId, aPath || '/' || c.Name, aDepth + 1
+                        FROM meta.ECClassDef c
+                            JOIN meta.ClassHasBaseClasses cbc ON cbc.SourceECInstanceId = c.ECInstanceId
+                            JOIN base_classes  ON aId = cbc.TargetECInstanceId
+                    ORDER BY 1
+                )
+            SELECT group_concat( DISTINCT p.Name)  from base_classes join meta.ECPropertyDef p on p.Class.id = aId
+        ))";
+
+        ECSqlStatement stmt;
+        ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, query));
+        stmt.BindText(1, "A", IECSqlBinder::MakeCopy::No);
+        ASSERT_EQ(BE_SQLITE_ROW, stmt.Step());
+        ASSERT_STREQ("a_prop,aa_prop,aaa_prop,aab_prop,ab_prop,aba_prop,abb_prop", stmt.GetValueText(0));
+    }

```

Comment:
I suggest you add this test case to the previous test to avoid adding some much identical setup code

Reply:
fixed